zkevm-circuits/src/evm_circuit.rs

```diff
@@ -1276,24 +316,83 @@ mod test {
             config: Self::Config,
             mut layouter: impl Layouter<F>,
         ) -> Result<(), Error> {
-            config.evm_circuit.load_fixed_tables(
+            config.evm_circuit.load_fixed_table(
+                &mut layouter,
+                self.fixed_table_tags.clone(),
+            )?;
+            config.load_txs(
+                &mut layouter,
+                &self.block.txs,
+                self.block.randomness,
+            )?;
+            config.load_rws(
+                &mut layouter,
+                &self.block.rws,
+                self.block.randomness,
+            )?;
+            config.load_bytecodes(
                 &mut layouter,
-                self.including_large_tables,
+                &self.block.bytecodes,
+                self.block.randomness,
             )?;
-            config
-                .evm_circuit
-                .load_rw_tables(&mut layouter, &self.operations)?;
+            config.evm_circuit.assign_block(&mut layouter, &self.block)
+        }
+    }
 
-            // load bytecode source from test sequence
-            let bytecode_table = assgin_byte_table_step(&self.execution_steps);
+    pub(crate) fn run_test_circuit<F: FieldExt>(
+        block: Block<F>,
+        fixed_table_tags: Vec<FixedTableTag>,
+    ) -> Result<(), Vec<VerifyFailure>> {
+        let log2_ceil = |n| {
+            u32::BITS - (n as u32).leading_zeros() - (n & (n - 1) == 0) as u32
+        };
```

Comment:
Since we are using `nightly` Rust, maybe can use [log](https://doc.rust-lang.org/std/primitive.u32.html#method.log) or [log2](https://doc.rust-lang.org/std/primitive.u32.html#method.log2)?

Reply:
I think using `u32::BITS - (n as u32).leading_zeros()` as `log2` would be enough (I am afraid introducing `#![feature(...)]` would also introduce other issues), furthermore the new feature `log2` seems to round down instead of ceiling, so I would prefer keep it unchanged here.